addon/utils/plugins/lists/delete-plugin.ts

```diff
@@ -149,21 +153,23 @@ export default class ListDeletePlugin implements DeletePlugin {
    */
   private mergeForwards(node: Node, editor: RawEditor) {
 
-    const mergeNode = this.getDeepestLastDescendant(node);
-    let cursorPosition = 0;
-    if (isTextNode(mergeNode) && mergeNode.textContent) {
-      cursorPosition = mergeNode.textContent.length;
-    }
+
     const targetNode = this.findNodeAfter(node, editor.rootNode);
     if (!targetNode) {
       // no node in front of us, no sense in looping 50 times
       this.hasChanged = true;
       return;
     }
+    const mergeNode = this.getDeepestLastDescendant(node);
+
+    const cursorPos = getCaretRect();
+    const selection = getWindowSelection();
+    selection.removeAllRanges();
     const nodeToMerge = this.getDeepestFirstDescendant(targetNode);
     this.mergeNodes(mergeNode, nodeToMerge);
 
-    this.repositionCursor(mergeNode, nodeToMerge, cursorPosition);
+    setCaretOnPoint(cursorPos.right, cursorPos.bottom - cursorPos.height/2 )
```

Comment:
I suppose the delete handler is responsible for making sure the editor state is up to date with this change.